onnxruntime/core/providers/dml/DmlExecutionProvider/src/Operators/DmlOperatorElementWise.cpp

```diff
@@ -560,13 +576,97 @@
+            TOperatorDesc opDesc = {};
+            opDesc.InputTensor = &inputDescs[OnnxInputIndex::inputIndex];
+            opDesc.ScaleTensor = &inputDescs[OnnxInputIndex::scaleIndex];
+            opDesc.ZeroPointTensor = &inputDescs[OnnxInputIndex::zeroPointIndex];
+            opDesc.OutputTensor = &outputDescs[0];
 
-        SetDmlOperatorDesc({ApiTraits::OperatorDescTraits<TOperatorDesc>::Type, &opDesc}, kernelInfo);
+            SetDmlOperatorDesc({ApiTraits::OperatorDescTraits<TOperatorDesc>::Type, &opDesc}, kernelInfo);
+        }
+        // Create a zero point tensor, since it's a required input by DML
+        else
+        {
+            auto inputSizes = m_inputTensorDescs[0].GetSizes();
+            TensorDesc intermediateOutputTensorDesc = TensorDesc(
+                    GetMlDataTypeFromDmlDataType(inputDataType),
+                    inputSizes,
+                    inputSizes,
+                    TensorAxis::DoNotCoerce,
+                    TensorAxis::W,
+                    TensorAxis::RightAligned,
+                    NchwDimensionCount, // minDimensionCount
+                    0 // guaranteedBaseOffsetAlignment
+                );
+            DML_TENSOR_DESC namedIntermediateOutputTensorDesc = intermediateOutputTensorDesc.GetDmlDesc();
+
+            // Create a tensor full of zeros
+            DML_FILL_VALUE_CONSTANT_OPERATOR_DESC zerosDesc = {};
+            zerosDesc.ValueDataType = inputDataType;
+            zerosDesc.OutputTensor = &namedIntermediateOutputTensorDesc;
+            const DML_OPERATOR_DESC opDesc1 = { DML_OPERATOR_FILL_VALUE_CONSTANT, &zerosDesc };
+
+            TOperatorDesc qLinearDesc = {};
+            qLinearDesc.InputTensor = &inputDescs[OnnxInputIndex::inputIndex];
+            qLinearDesc.ScaleTensor = &inputDescs[OnnxInputIndex::scaleIndex];
+            qLinearDesc.ZeroPointTensor = zerosDesc.OutputTensor;
+            qLinearDesc.OutputTensor = &outputDescs[0];
+            const DML_OPERATOR_DESC opDesc2 = { ApiTraits::OperatorDescTraits<TOperatorDesc>::Type, &qLinearDesc};
+
+            // Construct the graph
+            DML_INPUT_GRAPH_EDGE_DESC inputEdges[2];
+            DML_INTERMEDIATE_GRAPH_EDGE_DESC intermediateEdges[1];
+            DML_OUTPUT_GRAPH_EDGE_DESC outputEdges[1];
+
+            MLOperatorGraphDesc operatorGraphDesc = {};
+            operatorGraphDesc.nodeCount = 2;
+            const DML_OPERATOR_DESC* opDescs[] = {&opDesc1, &opDesc2};
+            operatorGraphDesc.nodesAsOpDesc = std::data(opDescs);
+
+            const uint32_t fillValueNodeIndex = 0;
+            const uint32_t dequantizeNodeIndex = 1;
+
+            // Input edges
+            DML_INPUT_GRAPH_EDGE_DESC inputToDequantizeEdge = {};
+            inputToDequantizeEdge.GraphInputIndex = OnnxInputIndex::inputIndex;
+            inputToDequantizeEdge.ToNodeIndex = dequantizeNodeIndex;
+            inputToDequantizeEdge.ToNodeInputIndex = 0;
+            inputEdges[0] = inputToDequantizeEdge;
```

Comment:
>inputEdges[0] = inputToDequantizeEdge; [](http://example.com/codeflow?start=12&length=38)

We can avoid the temporary copy by just writing into the DML_INPUT_GRAPH_EDGE_DESC that already exists anyway - shorter code, more efficient.

```
Foo foo = {}
foo.x = 42;
foos[0] = foo;
```

Becomes simply:

```
Foo& foo = foos[0]
foos.x = 42
```